tests/test_metadata_serialization.py

```diff
@@ -48,8 +50,61 @@ def wrapper(test_cls: "TestSerialization"):
     return real_decorator
 
 
+class TestSigned(Signed):
```

Comment:
I think creating a class just for testing it feels wrong. I would prefer using an existing Signed-derivative for testing these fields, but it's your choice.

Reply:
I removed one of the cases `one digit spec_version`, the others seem relevant to the checks we are making.

Reply:
I Will remove it, but as a side effect the `invalid_signed` tests will become longer.